app/customer_routes.py

```diff
@@ -0,0 +1,243 @@
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return make_response({"details": "Invalid data"}, 404)
+    
+    if request.method == "GET":
+        return make_response(video.video_to_json(),200)
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        if "name" not in form_data or "postal_code" not in form_data or "phone" not in form_data or bool(form_data) is False: 
+            return ({"details":"Invalid data"},400)
+
+        video.name = form_data["title"]
+        video.total_inventory = form_data["total_inventory"]
+        
+        db.session.commit()
+
+        return make_response(video.video_to_json(),200)
+
+    elif request.method == "DELETE":
+
+        db.session.delete(video)
+        db.session.commit()
+
+        return make_response({"id":video_id}, 200)
+
+
+@customers_bp.route("/<id>/rentals", methods=["GET"], strict_slashes = False)
+def get_rentals_by_customer(id):
+
+    customer = Customer.query.get(id)
+
+    if not customer:
+        return({"details":"Invalid data"},400)
+
+    rental_list =[]
+    for rental in customer.rentals:
+        video = Video.query.get(rental.video_id)
+        rental_list.append({
+            "release_date":video.release_date,
+            "title":video.title,
+            "due_date":rental.due_date
+            })
+
+    return jsonify(rental_list)
+
+
+@videos_bp.route("/<id>/rentals", methods=["GET"], strict_slashes = False)
+def get_rentals_by_video(id):
+
+    video= Video.query.get(id)
+
+    if not video:
+        return({"details":"Invalid data"},400)
+
+    rental_list =[]
+    for rental in video.rentals:
+        customer = Customer.query.get(rental.customer_id)
+        rental_list.append({
+            "name":customer.name,
+            "phone":customer.phone,
+            "postal_code":customer.postal_code,
+            "due_date":rental.due_date
+        })
+
+    return jsonify(rental_list)
+
+
+@rentals_bp.route("/check-out", methods=["POST"], strict_slashes = False)
+def check_out():
+
+    request_body = request.get_json()
+
+    if type(request_body["customer_id"]) is not int or "customer_id" not in request_body:
+        return({"details":"Invalid data"},400)
+
+    if type(request_body["video_id"]) is not int or "video_id" not in request_body:
+        return({"details":"Invalid data"},400)
+    
+    customer = Customer.query.get(request_body["customer_id"])
+    video = Video.query.get(request_body["video_id"])
+
+    if customer is None and video is None:
+        return {"error":"not found"}, 404
+
+    rental= Rental(
+        customer_id=customer.customer_id,
+        video_id=video.video_id,
+        due_date=datetime.utcnow()+timedelta(days=7)
+        )
+
+    if video.available_inventory == 0:
+        return({"details":"Invalid data"},400)
+    else:
+        video.available_inventory -= 1
+        customer.videos_checked_out_count += 1
+        db.session.add(rental)
+        db.session.commit()
```

Comment:
I strongly suggest pulling this "business logic" out of the `routes.py` file and into the models.  That way the routes only handles coordinating the requests and responses and the models can handle the logic of checking in and out a video.